LibreNMS/Data/Store/Rrd.php

```diff
@@ -132,6 +141,83 @@ public function close()
         }
     }
 
+    public function record(DataGroup $data)
+    {
+        foreach ($data->getDataSets() as $ds) {
+            $rrd = $this->fileName($data, $ds);
+            if (! $this->checkRrdExists($rrd)) {
+                $dir = "$this->rrd_dir/" . $data->getName();
+                if (! file_exists($dir)) {
+                    mkdir($dir);
+                }
```

Comment:
This breaks the remote RRD feature thingy right? Since the path might not be available on the server.

Can we use this?
>When you start RRDtool with the command line option '-' it waits for input via standard input (STDIN). With this feature you can improve performance by attaching RRDtool to another process (MRTG is one example) through a set of pipes. Over these pipes RRDtool accepts the same arguments as on the command line and some special commands like cd, **mkdir**, pwd, ls and quit.: